lxd/storage/drivers/driver_zfs_volumes.go

```diff
@@ -2332,6 +2332,14 @@ func (d *zfs) readonlySnapshot(vol Volume) (string, revert.Hook, error) {
 
 		mountFlags, mountOptions := filesystem.ResolveMountOptions(strings.Split(vol.ConfigBlockMountOptions(), ","))
 
+		// ext4 will replay the journal if the filesystem is dirty.
```

Comment:
Ive not seen this in other drivers (I could be wrong) bit I'm interested why this is specifically needed on zfs block vols?

Reply:
I have no idea TBH. I ran into issues when trying to mount the ro snapshot, and the logs suggested adding `noload`.

Reply:
It does make some sense. Where were you seeing that error? Have you tried other filesystems too?

I think its not an issue perhaps with lvm as those snapshots are writable, although we could probably do with doing the same thing on all of the block backed filesystem volumes.

Reply:
> It does make some sense. Where were you seeing that error? Have you tried other filesystems too?

I saw the error when `readonlySnapshot()` calls `TryMount()`. I haven't tried other filesystems yet.

Reply:
@tomponline I just tried migration with xfs and btrfs, and both fail at a different stage. They seem unrelated to the rsync change though. Need to fix them.